application_test.go

```diff
@@ -427,3 +427,20 @@ func TestApplicationConfiguration(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, ErrCodeNotFound, apiErr.ErrCode)
 }
+
+func TestWaitOnApplication(t *testing.T) {
+	endpoint := newFakeMarathonEndpoint(t, nil)
+	defer endpoint.Close()
+	client := endpoint.Client.(*marathonClient)
+
+	c := make(chan bool)
+	err := client.waitOnApplication(fakeAppName, c)
+	assert.NoError(t, err)
+
+	c = make(chan bool)
+	go func() {
+		c <- true
+	}()
```

Comment:
@gambol99, I think `atomicSwitch` was used just for synchronisation of two goroutines, so usage of blocking channel would be more go-stylish.

@timoreimann, You seems to be right about timer cleaning up after firing up an event.
I've changed implementation to using `After()`. I'll push changes if switching from `atomicSwitch` is ok.